twitch.py

```diff
@@ -605,7 +604,7 @@ def invalidate(self, *, auth: bool = False, integrity: bool = False):
 
 
 class Twitch:
-    def __init__(self, settings: Settings):
+    def __init__(self, settings: Settings, cli: bool = False):
```

Comment:
At this stage, the arguments are already parsed and merge with the `Settings`, so you should be using it's system to pass the CLI setting through there instead, instead of adding arguments to the client class.